pyro/distributions/torch_distribution.py

```diff
@@ -71,9 +71,20 @@ def reshape(self, sample_shape=torch.Size(), extra_event_dims=0):
         :param int extra_event_dims: The number of extra event dimensions that
             will be considered dependent.
         :return: A reshaped copy of this distribution.
-        :rtype: :class:`Reshape`
+        :rtype: :class:`ReshapedDistribution`
         """
-        return Reshape(self, sample_shape, extra_event_dims)
+        return ReshapedDistribution(self, sample_shape, extra_event_dims)
+
+    def mask(self, mask):
+        """
+        Masks a distribution by a zero-one tensor that is broadcastable to the
+        distributions ``batch_shape``.
+
+        :param Variable mask: A zero-one valued float tensor.
+        :return: A masked copy of this distribution.
+        :rtype: :class:`MaskedDistribution`
+        """
+        return MaskedDistribution(self, mask)
 
     def analytic_mean(self):
```

Comment:
nit: but masking probably kills analytic_mean etc.?

Reply:
Just as with `.sample()`, it is up to the user to ignore masked-out values of `.mean`.